pkg/virtctl/root.go

```diff
@@ -18,10 +18,18 @@ import (
 	"kubevirt.io/kubevirt/pkg/virtctl/vnc"
 )
 
+var programName string
+
 func NewVirtctlCommand() *cobra.Command {
+	elements := strings.Split(os.Args[0], "/")
+	programName = strings.Replace(elements[len(elements)-1], "-", " ", -1)
```

Comment:
Why do we need to replace "`-`" with "` `"?

Reply:
Because `virtctl` is linked to as kubectl-virt and the dash needs to be replaced:

```
─dhiller@dhiller-fedora-redhat ~/.krew/bin 
╰─$ ls -la
total 16
drwxr-xr-x. 2 dhiller dhiller 4096 Jul 11 14:36 .
drwxr-xr-x. 5 dhiller dhiller 4096 Jun  4 12:38 ..
lrwxrwxrwx. 1 dhiller dhiller  100 Jun  4 12:36 kubectl-krew -> /home/dhiller/.krew/store/krew/dc2f2e1ec8a0acb6f3e23580d4a8b38c44823e948c40342e13ff6e8e12edb15a/krew
lrwxrwxrwx. 1 dhiller dhiller  103 Jun 14 08:55 kubectl-virt -> /home/dhiller/.krew/store/virt/1ddf038ec1c60753d6d5b7cc37e43b74eac6a2bc47b9aa542eb9f7f41f7741e6/virtctl
```

Reply:
what if I download `virtctl` and call it `my-virtcl-v0.19`?

Reply:
Good point! Interestingly, I just tried that and it showed:
```
$ ./my-virtctl-v0.19 help                                                                    │
my virtctl v0.19 controls virtual machine related operations on your kubernetes cluster.       │
                                                                                               │
Available Commands:                                                                            │
  console      Connect to a console of a virtual machine instance.                             │
  expose       Expose a virtual machine instance, virtual machine, or virtual machine instance │
replica set as a new service.                                                                  │
  help         Help about any command                                                          │
  image-upload Upload a VM image to a PersistentVolumeClaim.                                   │
  restart      Restart a virtual machine.                                                      │
  start        Start a virtual machine.                                                        │
  stop         Stop a virtual machine.                                                         │
  version      Print the client and server version information.                                │
  vnc          Open a vnc connection to a virtual machine instance.                            │
                                                                                               │
Use "my virtctl v0.19 <command> --help" for more information about a given command.            │
Use "my virtctl v0.19 options" for a list of global command-line options (applies to all comman│
ds).                                                                                           │
```

So you are suggesting that we should guard specifically against the case where it is used in the krew context?

Reply:
Is there a way to find out if it is called as a kubectl/krew plugin? Maybe an environment variable?

Reply:
Not that I know of. I didn't find anything in the docs.

What we know is (as we are in control of how our plugin is supposed to be named, that the call should have a basename of `kubectl-virt`. We could see if it equals on that and if it doesn't we just use the basename.

Reply:
That we know because the `kubectl` plugin mechanism works this way.

Reply:
yep, didn't find anything either. I guess if the basename is exactly `kubectl-virt`, then we can do the magic and modify the basename and print `kubectl virt`. Otherwise I would be inclined to just write `virtctl` always, but that differes from tool to tool what it does in this case (e.g. `ls` shows what you typed exactly, whereas `git` always shows `git`).

Reply:
I think I remember that it was a best practice to hardcode the name of your tool, so that you still know what the tool is you are using, no matter how the binary or symlink is named.

Reply:
Thins would mean that we would stick with the original name `virtctl`, right? 

But this would mean to confuse the user if he has installed virtctl via krew and then using the help from the `kubectl virt` perspective (which is the original issue):

```
$ kubectl virt help
virtctl controls virtual machine related operations on your kubernetes cluster.

Available Commands:
  console      Connect to a console of a virtual machine instance.
  expose       Expose a virtual machine instance, virtual machine, or virtual machine instance replica set as a new service.
  help         Help about any command
  image-upload Upload a VM image to a PersistentVolumeClaim.
  restart      Restart a virtual machine.
  start        Start a virtual machine.
  stop         Stop a virtual machine.
  version      Print the client and server version information.
  vnc          Open a vnc connection to a virtual machine instance.

Use "virtctl <command> --help" for more information about a given command.
Use "virtctl options" for a list of global command-line options (applies to all commands).
```

Maybe we could improve the `kubectl virt` scenario by *only* using `kubectl virt` if the basename is `kubectl-virt`?

Reply:
Oh, you proposed that already, overread your previous comment.

OK, will change it to only display `kubectl virt` *IIf* the basename is `kubectl-virt`.